web/json_options.go

```diff
@@ -80,7 +81,11 @@ func defaultJSONOptions() *jsonOptions {
 			{"01-02-2006", false},
 			{"01/02/2006", false},
 			{"01/02/06", false},
-			{SGNLUnixSec, false},        // Unix timestamp representing seconds since 1970-01-01 00:00:00 UTC.
+
+			// The ordering of SGNLUnixMilli and SGNLUnixSec is important.
```

Comment:
Can you elaborate as to why this is the case? I presume there's a risk of `SGNLUnixMilli` being parsed as a `SGNLUnixSec` if the order is reversed, but I think this warrants an explanation

Reply:
Also, regardless of the ordering there's going to be some overlap. Any value provided <= `999999999` is valid for both `time.Unix` and `time.UnixMilli`.  If a user specifies `999999999` with the default JSON options, it's not apparent to the user if this will be represented as `1970-01-12 13:46:39.999 +0000 UTC` or `2001-09-09 01:46:39 +0000 UTC`.

I think this is a very niche edge case we probably won't encounter, but at the very least I think this is worth documenting (e.g. if `SGNLUnixMilli` and `SGNLUnixSec` are both enabled, any unix timestamp <= `999999999` is assumed to be provided in seconds. Otherwise, this is assumed to be provided as milliseconds.).

Alternatively, we could explore adding some validation to return an error if a user tries to specify both of these at the same time (assuming there's not a single use case that relies on both).

Reply:
I agree. 

Values above `999999999`, e.g. `1706041056000`, are also successfully parsed by `SGNLUnixSec` as well, although to an absurd timestamp like `time.Date(56032, time.April, 15, 13, 20, 0, 0, time.UTC)`.

I was just testing this edge case and looking for alternative solutions.

Unix timestamp represented as seconds is the most commonly used format from what I read. To avoid any confusion, based on popularity, I'd propose we retain `SGNLUnixSec` in default options. 

`SGNLUnixMilli` can be used on an as-needed basis by adapters.

WDYT?

Reply:
Good point, I misinterpreted the spec that mentioned that `nsec` needs to be in the range `[0, 999999999]` for calls to `Unix(...)`. 

Since it seems like any value that can be parsed successfully as either `UnixMilli` or `Unix` can also be parsed as the other, we should never have a situation where we define both of these values.

WDYT about adding validation in `ConvertJSONObjectList` to prevent framework users from specifying both of these? This behavior isn't immediately obvious to a user of the framework, and I worry without this we either ship a buggy adapter or a third party runs into issues creating an adapter because of this. As far as I'm aware we don't have any other formats currently that conflict in this manner.

I'm good with leaving `SGNLUnixSec` as the default option.

Reply:
Sounds like a plan! 
I'll take up the validation changes in a follow up PR if there's no objections.

Reply:
I assume you're looking at cutting a release right after this is merged so you're able to use this in the adapter you're developing? IMO this validation should be in place before we cut any new releases since this is a public repo that a user could theoretically update to immediately - so my preference would be to include this in this PR since this doesn't seem feature complete without it

Reply:
I'm thinking of adding validation here.

https://github.com/SGNL-ai/adapter-framework/blob/8fe08dd7237fdcade18be2dc3bbf9c53e69c993b/web/json_options.go#L139

As this function does not already return an error, we'd need to update the function signature to return an error. This would result in changes across several adapters increasing the scope of this story.

As SGNL's the only one developing adapters right now, I think it'd be safe to postpone adding validation as a part of this PR.

Reply:
Yeah, as discussed during the eng weekly that's good with me as long as we add the proper comments

Reply:
Added a note on using both unix datetime formats.